tests/functional/ctu_failure/__init__.py

```diff
@@ -0,0 +1,35 @@
+# coding=utf-8
+# -----------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -----------------------------------------------------------------------------
+
+"""Setup for the test package ctu."""
+
+import os
+import shutil
+
+from libtest import env
+
+# Test workspace should be initialized in this module.
+TEST_WORKSPACE = None
+
+
+def setup_package():
+    """Setup the environment for testing ctu."""
+
+    global TEST_WORKSPACE
+    TEST_WORKSPACE = env.get_workspace('ctu_failure')
```

Comment:
We have a **lot** of subdirectories under `functional` already. Can't the test module file of this test package instead go as a new python module in the `ctu` folder?

Reply:
If I understand the structure of the test dirs correctly then each dir represent a test suite with its own setup and optionally files for the test project. And I think, the ctu_failure directory perfectly fits into this structure.
```
   ▾ delete_runs/            
       __init__.py           
       test_delete_runs.py   
   ▾ detection_status/       
       __init__.py           
       test_detection_status.
   ▾ diff/                   
       __init__.py           
       test_diff.py          
   ▾ func_template/          
       template__init__.py   
       template_test.py      
   ▾ instance_manager/       
       __init__.py           
       test_instances.py     
```

Reply:
I don't think there ever was any structural decision on the functional test folder. The [suppress tests before review status was implemented](https://github.com/Ericsson/codechecker/tree/v5.8/tests/functional/suppress), for example, also tested multiple features at once.

The current layout of the test suite better represents testing individual features or feature sets of CodeChecker. Analyzer handling, analyzer result handling, result querying (report viewer api), diffing, run deletion, product management, etc.

"CTU failure" is not a feature set, it is a new, special test case for CTU. Of course, due to this test being related to analyzer handling, it *could* go into the `analyze` folder, but `CTU` is a bigger, and thus more powerful category.

Reply:
Structurally, nosetest test on the module level, the package level things only handle analysis execution, result storage and server setup.

For example, the current layout is also bad, as every single damn test package runs a full analysis, sometimes on a big project, but at least where it wasn't neccessarry, we cut down on using a simpler project only, thus further quickening up the tests without losing functionality.

Reply:
Ok, I moved the ctu_failure dir under ctu.